Src/main.c

```diff
@@ -106,8 +122,28 @@ int main(void)
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+
   while (1)
   {
+    /*start and wait for conversiono (wait 2000 ms max)*/
+    if(HAL_OK == HAL_ADC_PollForConversion(&hadc, 2000)){
+      float raw_data = HAL_ADC_GetValue(&hadc); /*get converted digital value - we are assuming ADC will be properly configured to the potentiometer*/
+
+      /*Use digital value to determine duty cycle*/
+      /*Total period is 20 ms*/
+      /*Changing the pulse width (between 1ms to 2ms) will change the angle of the shaft of the "servo motor"*/
+      /*change from 1.5 ms to 1ms will rotate from 0 degrees to -90 degrees. 1.5ms to 2 ms will rotate 0 degrees to 90 degrees*/
+      float percentage = (raw_data/(pow(2, ADC_RESOLUTION)-1)); //percentage = ADC reading/Resolution of ADC - Resolution od ADC = 2^(# of bits) - ADC_RESOLUTION = 12
+      float on_time = (percentage * (ON_TIME_UPPER_LIMIT - ON_TIME_LOWER_LIMIT)) + ON_TIME_LOWER_LIMIT; //unit is ms
+      float pulse_width = (on_time/20) * 60000;
```

Comment:
Could you use constants here? In code it is sometimes better to use constants instead of writing numbers to improve readability, etc. 

Something like `TIMER_PERIOD` will do.